Mage.Sets/src/mage/cards/k/KyloxsVoltstrider.java

```diff
@@ -0,0 +1,166 @@
+                new CollectEvidenceCost(6, true)));
+
+        // Whenever Kylox's Voltstrider attacks, you may cast an instant or sorcery spell from among cards exiled with it.
+        // If that spell would be put into a graveyard, put it on the bottom of its owner's library instead.
+        this.addAbility(new AttacksTriggeredAbility(new KyloxsVoltstriderEffect(), true));
+
+        // Crew 2
+        this.addAbility(new CrewAbility(2));
+
+    }
+
+    private KyloxsVoltstrider(final KyloxsVoltstrider card) {
+        super(card);
+    }
+
+    @Override
+    public KyloxsVoltstrider copy() {
+        return new KyloxsVoltstrider(this);
+    }
+}
+
+class KyloxsVoltstriderEffect extends OneShotEffect {
+
+    KyloxsVoltstriderEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "you may cast an instant or sorcery spell from among cards exiled with it. " +
+                "If that spell would be put into a graveyard, put it on the bottom of its owner's library instead.";
+    }
+
+    private KyloxsVoltstriderEffect(final KyloxsVoltstriderEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KyloxsVoltstriderEffect copy() {
+        return new KyloxsVoltstriderEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        UUID exileId = CardUtil.getExileZoneId(game, source.getSourceId(),
+                game.getState().getZoneChangeCounter(source.getSourceId()));
+        ExileZone exileZone = game.getExile().getExileZone(exileId);
+        if (controller == null || exileZone == null) {
+            return false;
+        }
+
+        TargetCard target = new TargetCardInExile(StaticFilters.FILTER_CARD_INSTANT_OR_SORCERY, exileId);
+        target.withNotTarget(true);
+        if (!target.canChoose(source.getControllerId(), source, game)) {
+            return true;
+        }
+        controller.chooseTarget(outcome, target, source, game);
+        Card card = game.getCard(target.getFirstTarget());
+        if (card != null) {
+            game.getState().setValue("PlayFromNotOwnHandZone" + card.getId(), Boolean.TRUE);
+            controller.cast(controller.chooseAbilityForCast(card, game, false),
+                    game, false, new ApprovingObject(source, game));
+            game.getState().setValue("PlayFromNotOwnHandZone" + card.getId(), null);
+
+            // If that spell would be put into a graveyard, put it on the bottom of its owner's library instead.
+            MageObjectReference mor = new MageObjectReference(card, game);
+            ContinuousEffect effect = new KyloxsVoltstriderReplacementEffect(mor);
+            game.addEffect(effect, source);
+        }
+        return true;
+    }
+}
+
+class KyloxsVoltstriderReplacementEffect extends ReplacementEffectImpl {
+
+    private final MageObjectReference mor;
```

Comment:
what's this based on? can use standard fixed target without needing MOR (see ThatSpellGraveyardExileReplacementEffect for example), though I guess this is fine

Reply:
Initially I based it on `ThatSpellGraveyardExileReplacementEffect`, but the spell was still going to the graveyard. So I looked around for similar cards and found Quintorious Loremaster. I've now managed to get the original working. It seems that the zone change offset needs to be 0 rather than 1 for this.

Reply:
Well that's disconcerting... why would the zcc be different between the two effects?

Reply:
Oh never mind, it's because of where the fixed target is initialized (before vs after cast)